### PR TITLE
[WIP] Randomize fuzzing params with prob

### PR DESCRIPTION
**Pull request by jiradeto**
This PR randomly adjusts AFL's fuzzing parameters which typically are constants. Basically at the beginning of the fuzzing cycle, we repeatedly randomize parameter's value with some probabilities that can be set with env variable `AFL_RP_PROB`.

### CODE REVIEW

File: afl-fuzz.c [hunk 1]
```diff
@@ -1274,6 +1286,12 @@ double rand_double()
   return rnd / max;
 }
 
+int rand_int_in_range(int low, int high) {
+    double rnd = rand() / (1.0 + RAND_MAX);
+    int range = high - low + 1;
+    int rnd_scaled = (rnd * range) + low;
+    return rnd_scaled;
```
Comment (wuestholz) [inline afl-fuzz.c]: @jiradeto Could we simply use `return low + UR(range)` here?

**Comment by wuestholz** [comment]
@jiradeto Thanks! Somehow I must have missed this PR... It looks like it's going in the right direction. Have you already tested this a bit? If so, you can probably start extending it to cover more parameters.

**Comment by jiradeto** [comment]
@wuestholz, I have started the initial experiment with just a single parameter tweak `HAVOC_STACK_POW2`. The results are available for [bloaty_sqlite](https://eloquent-bohr-97b0cb.netlify.app/results/2021-07-28-initial_random_params/bloaty_sqlite/index.html) and [libjpeg_libxml](https://eloquent-bohr-97b0cb.netlify.app/results/2021-07-28-initial_random_params/libjpeg_libxml/index.html). 

And yeah, if this PR looks good I will add more fuzzing parameters and start experimenting shortly.

**Comment by wuestholz** [comment]
@jiradeto Oh, wow! The differences are quite big and it looks like in some benchmarks the randomization helps.

What value did you use for `AFL_RP_PROB`?

**Comment by jiradeto** [comment]
@wuestholz, sorry I gave you the wrong experiment. The one I shown you is the initial experiment with `AFL_RP_PROB = 0` just to confirm that there is no issue with this PR. The differences shown is solely because of the changes from https://github.com/Practical-Formal-Methods/AFL-public/pull/6 in which this PR is based.

I just pushed the latest changes which adds more parameters with a set of random values from our discussion.

**Comment by wuestholz** [comment]
@jiradeto Oh, I see. :) No problem! Yeah, let's see what the effect is once we enable this. I might start with `AFL_RP_PROB=20`.

File: afl-fuzz.c [hunk 2]
```diff
@@ -8291,6 +8332,19 @@ int main(int argc, char** argv) {
       if (!disable_weighted_random_selection)
         mark_selected_inputs();
 
+      if (!disable_randomized_fuzzing_params) {
+        // randomize fuzzing params with probabilities
+        OKF("WEASDASDASD %d", randomize_parameters_prob);
```
Comment (wuestholz) [inline afl-fuzz.c]: @jiradeto Please remove the log output. :)

**Pull request merged**